cpp/velox/substrait/SubstraitToVeloxPlan.cc

```diff
@@ -1272,26 +1272,30 @@ core::PlanNodePtr SubstraitToVeloxPlanConverter::toVeloxPlan(const ::substrait::
     SubstraitParser::parseColumnTypes(baseSchema, columnTypes);
   }
 
-  // Velox requires Filter Pushdown must being enabled.
-  bool filterPushdownEnabled = true;
   auto names = colNameList;
   auto types = veloxTypeList;
-  auto dataColumns = ROW(std::move(names), std::move(types));
+  // The columns we project from the file.
+  auto baseSchema = ROW(std::move(names), std::move(types));
+  // The columns present in the table, if not available default to the baseSchema.
+  auto tableSchema = splitInfo->tableSchema ? splitInfo->tableSchema : baseSchema;
```

Comment:
> we need to pass the full table schema to the HiveTableHandle,

Is table schema set only when mapping by index?

Reply:
For now, when I add support for TextFiles it will be set when the file format is text as well.

Reply:
> if the table schema contains types that aren't supported in the Substrait, sending the table schema would cause the query to fail

Can we say that setting the table schema can lead to unexpected failure even when matching by index?

Reply:
> I'm not sure which is preferrable (falling back to Vanilla Spark more often or failing in what I'm assuming will be rare cases).

If possible, I’d prefer to fallback rather than fail for unsupported schema.